src/server/storage/cron/CleanupAbandonedSignups.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import logger from '../../../imports/logger'
+import Config from '../../server.config'
+
+class CleanupAbandonedSignups {
+  schedule = null
+  model = null
+  logger = null
+
+  get name() {
+    return 'storage/cleanup_abandoned_signups'
+  }
+
+  constructor(Config, storage, logger) {
+    const { storageCleanupCron } = Config
+
+    this.logger = logger
+    this.schedule = storageCleanupCron
+    this.model = storage.model
+  }
+
+  async execute() {
+    const { model, logger } = this
+
+    try {
+      // TODO: need to check
+      await model
+        .find({
+          torusProvider: { $not: { $type: 'string', $ne: '' } }
```

Comment:
i think some time condition (older than xxx) should also be added here
as the task schedule, it also should be defined by env var / config